src/lib/es2015.iterable.d.ts

```diff
@@ -27,6 +27,10 @@ interface Iterator<T, TReturn = any, TNext = undefined> {
     throw?(e?: any): IteratorResult<T, TReturn>;
 }
 
+interface BuiltinIterator<T, TReturn = void, TNext = undefined> extends Iterator<T, TReturn, TNext> {

```

Comment:
It's possible this may need to be
```ts
interface BuiltinIterator<T, TReturn = any, TNext = any> extends Iterator<T, TReturn, TNext> {
```
based on my findings in #58243, or `BuiltinIterator` shouldn't supply defaults at all.

Reply:
For now I've opted to use the weaker defaults.